api/envoy/type/http/v3/path_transformation.proto

```diff
@@ -49,7 +61,13 @@ message PathTransformation {
 
       // Enable merging adjacent slashes.
       MergeSlashes merge_slashes = 3;
+
+      // Enable unescaping escaped slashes.
+      UnescapeSlashes unescape_slashes = 4;
```

Comment:
One aspect of this design is we're not going to find it easier to change defaults without at least a boolean inside these empty messagese.

Reply:
What do you mean by change defaults?

Reply:
Hmm, yes I think this may be a problem. Right now we use the new API iff the [path_normalization_options](https://github.com/envoyproxy/envoy/blob/7cc10b03bcbff0367daaa8bd9b228eee9871d0e3/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L704) in the HCM config is set. Otherwise we use the old settings such as normalize_path, merge_slashes, etc. Since the old values are `bool` there is no way to distinguish between the "not set" and "disabled" cases.
What is the best approach here Harvey? Should we deprecate and not even consider old fields after deprecation period with some runtime override? This seems dangerous.

Reply:
> What do you mean by change defaults?

For example if we want the `merge_slashes` to be enabled by default in Envoy, with just empty messages there is no way for the operator to say I want to override Envoy default and **not** do slash merging

Reply:
Yep, that's a big deal. I think users will need to set the new message options until a minor window version has passed and we stop supporting the old field, CC @adisuissa.

My other observation is that in the new API, maybe we're setting ourselves up for the same problem. The new `merge_slashes` field can either be unset or set. If we want to make the default that when we have a `PathTransformation` in some use case that it is true if not set, and in another false if not set, we can't do that, because we don't offer hte user a way of overriding a true default. I think the solution is to include an `enable` boolean inside something like `MergeSlashes`.

Reply:
Can we simplify it by using a runtime guard? If the runtime is disabled, we will totally ignore the new field and go back to the old behavior.

Reply:
I don't think we can rely on runtime enablement forever, this is something that we typically do for new or transitioning features.

Reply:
I'm looking at this thread with fresh eyes on the PR. I think that if set `PathTransformation` at all, we can say "we now ignore all the old fields for path normalization" and in that case, all normalizations are disabled. Then we enable the ones we want using `operations`. When opting for `PathTransformation`, you make a conscious decision to configure sensible options.

If `PathTransformation` is not set, we get whatever legacy normalizations exist by default.

@yanavlasov @chaoqin-li1123 does that make sense?

Reply:
That's what the current implementation do, ignore the old field if the new field is set.

Reply:
OK, makes sense, thanks!